src/com/googlecode/jmxtrans/model/naming/KeyUtils.java

```diff
@@ -160,7 +160,7 @@ public static String getConcatedTypeNameValues(List<String> typeNames, String ty
 		StringBuilder sb = new StringBuilder();
 		for (String key : typeNames) {
 			String result = typeNameValueMap.get(key);
-			if (result != null) {
+			if (result != null && !StringUtils.isBlank(result)) {
```

Comment:
I wonder more how the value gets put into typeNameValueMap as null. I'd rather fix it there.